app/javascript/tags/Tag.jsx

```diff
@@ -0,0 +1,97 @@
+import { h } from 'preact';
+import { useState } from 'preact/hooks';
+import PropTypes from 'prop-types';
+/* global browserStoreCache */
+
+/**
+ * Renders the updated buttons component for a given tag card
+ * @param {number} props.id The id of the tag
+ * @param {string} props.name The name of the tag
+ * @param {boolean} props.isFollowing Whether the user is following the tag
+ * @param {string} props.isHidden Whether the tag is hidden
+ *
+ * @returns Updates the given Tag buttons (Follow and Hide) with the correct labels, buttons and actions.
+ */
+export const Tag = ({ id, name, isFollowing, isHidden }) => {
```

Comment:
As per my context on the preact component in the first comment/description, I'd like to abstract out a single button into its own component. For example: we'd have a Button component that we'd re-use to manage the state of the 'Follow' button and the 'Hide' button. This component would then only manage the interaction between those two buttons. 

Thereafter, we could possibly re-use this button component for some of the other follow states in `followButtons.js`. I did not implement this right now because the main priority is to get this functionality out. If you've reviewed other work of mine as part of this feature, you'd have heard me mention that I plan to write up follow up issues to address the technical debt and also to polish up some of the code.